packages/kit/src/runtime/client/client.js

```diff
@@ -677,6 +677,18 @@ export function create_client({ target, base }) {
 				try {
 					lock_fetch();
 					data = (await node.shared.load.call(null, load_input)) ?? null;
+					const not_object = typeof data !== 'object' && data != null;
+					if (not_object || Array.isArray(data) || data instanceof Response) {
+						throw new Error(
+							`a load function related to route '${route.id}' returned ${
+								not_object
+									? `a ${typeof data}`
+									: data instanceof Response
+									? 'a Response object'
+									: 'an array'
+							}, but must return a plain object (e.g '{ data: ...}')`
+						);
+					}
```

Comment:
Can we do this check only in dev mode?

Separately, can we be stricter about enforcing that this is a POJO? Something like `Object.getPrototypeOf(data) === Object.prototype`. (This would also cover the array and `Response` cases.) (Would that break any proxy stuff? I'm not sure whether we're doing any here.) We could also do this on the server.

Reply:
tested this out, a proxy still satisfies this check, so we should be able to use this check instead